The setuptools_scm version restriction isn't relevant and causes the build to
fail, so drop it.

--- a/setup.py
+++ b/setup.py
@@ -109,7 +109,7 @@
     # for example:
     # $ pip install -e .[dev,test]
     extras_require={'doc': read_requirements('doc')},
-    setup_requires=['setuptools_scm<6.0'],
+    setup_requires=['setuptools_scm'],
     use_scm_version=True,
 
     # If there are data files included in your packages that need to be
